src/Listener/UpdateMentionsMetadataWhenVisible.php

```diff
@@ -41,19 +41,29 @@ public function __construct(NotificationSyncer $notifications)
      */
     public function handle($event)
     {
-        $content = $event->post->parsedContent;
-
         $this->syncUserMentions(
             $event->post,
-            Utils::getAttributeValues($content, 'USERMENTION', 'id')
+            $this->mentionValue($event, 'users')
         );
 
         $this->syncPostMentions(
             $event->post,
-            Utils::getAttributeValues($content, 'POSTMENTION', 'id')
+            $this->mentionValue($event, 'posts')
         );
     }
 
+    protected function mentionValue($event, string $type)
+    {
+        $content = $event->post->parsedContent;
+        $tag = $type === 'users' ? 'USERMENTION' : 'POSTMENTION';
+
+        if ($event->actor->can("mention.$type", $event->post->discussion)) {
```

Comment:
Due to https://github.com/flarum/core/blob/bbd815a9abba837324c539874f8e04cf9c1a214f/src/User/UserServiceProvider.php#L106, this might not work as intended. I'd wait for a fix there rather than changing stuff here though.